GG/GG/Font.h

```diff
@@ -206,6 +209,22 @@ class GG_API Font
 
         virtual ~TextElement(); ///< Virtual dtor.
 
+        /** Rebind to \p whole_text, by binding the SubString data member text to \p whole_text.
+
+            Binding to a new \p whole_text is very fast compared to re-parsing the entire \p
+            whole_text and allows TextElements of TextElementType TEXT to be changed quickly if it
+            is known that the parse would be the same.
+
+            One use case is copying or copy constructing a TextControl.
+
+            Another use case is constructing a string and TextElement vector pair with
```

Comment:
tldr: Would changing the name `Bind()` to something more explicit like `AttachTo(const string&)` fix the problem?

I think you are correct that this is too low level to immediately see the relevance.  `Bind()` does what its name implies and binds, reassign an internal pointer of the `Substring` to a new string.  `TextElement` which is a composite containing a `Substring` needs bind its `Substring`s members to new strings.

I would remove the cases I added to the function comment, because `Bind()` is general and will be needed for all cases where a `Substring` needs to point to a new string, because they are tightly coupled.  

The motivating case for `TextAndElementsAssembler` was from `StatisticIcon::DoLayout()` and it might clarify the use case for `Bind()`.  But, again this is two steps removed from the `Bind()` function on its own.

`StatisticIcon::DoLayout()` became a bottleneck in improving `FleetWnd`.   I will explain that here incase that helps you decide how to comment `Bind()`.  `DoLayout()` generates the icon plus number used in many places in freeorion: `FleetWnd`, `PlanetTab`, `MapWnd`'s tab bar, etc.  For the `FleetWnd` alone avoiding re-parsing the XML saves (6 per fleet) * (6 per ship in fleet) of the parsing cost each time.  

Here is a section of the code from PR #992, which uses `TextAndElementsAssembler`, which in turn uses `Bind()`.
      
        `// Precompute text elements
    GG::Font::TextAndElementsAssembler text_elements(*ClientUI::GetFont());
    text_elements.AddOpenTag(ValueColor(0))
        .AddText(DoubleToString(m_values[0], m_digits[0], m_show_signs[0]))
        .AddCloseTag("rgba");

        // Create new label in correct place.
        m_text = new CUILabel(text_elements.Text(), text_elements.Elements(),
                              format, GG::NO_WND_FLAGS,
                              text_ul.x, text_ul.y, text_lr.x - text_ul.x, text_lr.y - text_ul.y);
        `

Would changing the name `Bind()` to something more explicit like `AttachTo(const string&)` fix the problem?